code/game/objects/items/devices/scanners.dm

```diff
@@ -673,11 +673,20 @@ NANITE SCANNER
 	var/list/discovered = list() //hit a dna console to update the scanners database
 
 /obj/item/sequence_scanner/attack(mob/living/M, mob/living/carbon/human/user)
-	user.visible_message("<span class='notice'>[user] has analyzed [M]'s genetic sequence.</span>")
+	if (!M.has_trait(TRAIT_RADIMMUNE) && !M.has_trait(TRAIT_BADDNA)) //no scanning if its a husk or DNA-less Species
+		user.visible_message("<span class='notice'>[user] has analyzed [M]'s genetic sequence.</span>")
 
-	add_fingerprint(user)
+		add_fingerprint(user)

```

Comment:
Why not just move this outside the if statement, and remove from both branches?

Reply:
didnt think of doing that, il fix it.